src/App.tsx

```diff
@@ -1,61 +1,161 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [visibleTodos, setTodos] = useState(todos);
+  const [username, setUsername] = useState('0');
+  const [isNoTitle, setIsNoTitle] = useState(false);
+  const [isNoUser, setIsNoUser] = useState(false);
+
+  const largestId = Math.max(...todos.map(todo => todo.id));
+
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>
+  | React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+
+    switch (name) {
+      case 'title':
+        if (value.length !== 0) {
+          setIsNoTitle(false);
+        }
+
+        // eslint-disable-next-line no-case-declarations
+        const newValue = value.trimStart().replace(/[^a-zA-Z0-9 ]/g, '');
```

Comment:
why you need this 🙃 ?

Reply:
I tried to do this optional task :) (* Optional) Allow entering only letters (ru and en), digits, and spaces in the title field. Just remove any other characters from the title.